token/services/interop/exchange/scanner.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package exchange
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"time"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/interop/encoding"
+	fabric2 "github.com/hyperledger-labs/fabric-token-sdk/token/services/network/fabric"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/vault/translator"
+	"go.uber.org/zap/zapcore"
+)
+
+// ScanForPreImage scans for a preimage of the passed image, taking into account the timeout
```

Comment:
Please, clarify that this functions scans the ledger and not the vault. It is therefore a heavy function. Just to alert the developer.